### PR TITLE
Task: Add project health

### PR DESCRIPTION
**Pull request by kynv-3493**
## Related Tickets
- (https://edu-redmine.sun-asterisk.vn/issues/67923)

## WHAT (optional)
- Thêm tình trạng sức khỏe của dự án

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)

[Screencast from 12-09-2023 15:03:00.webm](https://github.com/awesome-academy/project_management/assets/117953989/c9ac5b9a-8a62-48b1-92c5-491a764159e1)

### CODE REVIEW

**Comment by kynv-0324** [comment]
ready

**Comment by kynv-3493** [comment]
ready

File: app/controllers/health_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class HealthController < ApplicationController
+  before_action :logged_in_user, :check_permission, only: %i(new create)
+
+  def new
+    @health_items = HealthItem.where is_enabled: true
```
Comment (ngocvt-0484) [inline app/controllers/health_controller.rb]: viết scope cho where rồi gọi e nhé

File: app/controllers/health_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  ensure
+    redirect_to resources_path
+  end
+
+  private
+  def create_project_health_items
+    params.require(:project_id)
+    params.require(:health_items)
+
+    ActiveRecord::Base.transaction do
```
Comment (ngocvt-0484) [inline app/controllers/health_controller.rb:20]: làm kiểu này dc k e? modify cái params
`if @project.save project_feature_params
...
else
..
end


def project_feature_params
   project_feature_params = params.require(:..).permit(....)
   project_feature_params["project"]["project_feature_attributes"].values.each do |feature_params|
    feature_params.merge!(project_id: 1, month: 2, year: 3)
   end
   project_feature_params
end`

Comment (ngocvt-0484) [reply app/controllers/health_controller.rb:20]:
```
if @project.save project_feature_params
...
else
..
end


def project_feature_params
   project_feature_params = params.require(:..).permit(....)
   project_feature_params["project"]["project_feature_attributes"].values.each do |feature_params|
    feature_params.merge!(project_id: 1, month: 2, year: 3)
   end
   project_feature_params
end
```

Comment (ngocvt-0484) [reply app/controllers/health_controller.rb:20]: làm theo kiểu nested attibute ấy

File: config/routes.rb [hunk 3]
```diff
@@ -16,5 +16,6 @@
     resources :users, only: %i(edit update)
     resources :reports, only: %i(new create index destroy show)
     resources :release_plans, only: %i(edit update)
+    resources :health, only: [:new, :create]
```
Comment (ngocvt-0484) [inline config/routes.rb]: dùng %i

**Pull request closed without merging**